src/expression.d

```diff
@@ -11069,6 +11069,18 @@ extern (C++) final class DeleteExp : UnaExp
                 return new ErrorExp();
         }
 
+        // unsafe
+        if (!sc.intypeof && sc.func)
+        {
+            if (sc.func.setUnsafe())
+            {
+                error("%s is not @safe but is used in @safe function %s", toChars(), sc.func.toChars());
+                err = true;
+            }
+        }
+        if (err)
```

Comment:
I'd use a variable `result` initiated with `this`, then assign it from `new ErrorExp` in case of error. No need for a Boolean.